### PR TITLE
Simplify and correct include validation

### PR DESCRIPTION
**Pull request by watsonator**
* Allow for parent directories to be included
* Disallow absolute paths as includes (these were being misinterpreted)
* Expanded unit tests to match

### CODE REVIEW

**Comment by ThiefMaster** [comment]
Is there any real gain from being able to use `foo/../bar.html` in a template path? This seems more like something that would just encourage some people to write less readable code.

File: jinja2/loaders.py [hunk 1]
```diff
@@ -20,20 +21,14 @@
...
-           piece == path.pardir:
-            raise TemplateNotFound(template)
-        elif piece and piece != '.':
-            pieces.append(piece)
+    """Split a path into segments and perform a sanity check.  If it detects 
+    an absolute path it will raise a `TemplateNotFound` error."""
+    template = os.path.normpath(template)
+    altsep = os.altsep if os.altsep else ''
+    pieces = re.split(r'[%s%s]' % (os.sep, altsep), template)
+    if pieces[0] == '': raise TemplateNotFound(template)
```
Comment (ThiefMaster) [inline jinja2/loaders.py]:
```
if not pieces[0]:
    raise ...
```

**Comment by davidism** [comment]
I don't like that this changes how template paths are split, see my comments on #558.  Template paths are not filesystem paths, they just happen to look similar.

**Comment by ThiefMaster** [comment]
It also seems very confusing when you have more complex loader setups where you don't simply load templates from a certain filesystem location.

Imagine you have a loader that takes care of `/foo/*` and one that loads everything else. Would `/foo/../bar.html` be `/bar.html` for the `foo` loader, an invalid path (trying to go above the root of `/foo/`) or `/bar.html` in the other loader?

**Pull request closed without merging**